torch/fx/symbolic_trace.py

```diff
@@ -431,7 +431,11 @@ def forward(*args, **kwargs):
 
         self.submodule_paths = None
 
-        return self.graph
+        graph = self.graph
+        graph.tracer = self
+        self.graph = None
```

Comment:
I ran into pickling issues due to this cyclical reference so I added this to break the cycle when tracing is done.

Reply:
seems fine but add a comment?

Reply:
actually what was the error? circular references should be supported by the pickler.

Reply:
Something something `SimpleTest` could not be pickled. I don't have conclusive proof that it was because of this circular reference, but I did notice that pickling works fine without it.

Reply:
After making other changes (like moving those test classes into a separate module) I'm getting a different error. I uploaded a new version of the PR with all the other changes but I'll keep looking into this.

Reply:
Wait actually, we intentionally do not package the graph in `__reduce_package__` for `GraphModule`:
```
    def __reduce_package__(self, exporter: PackageExporter):
        generated_module_name = f'fx-generated._{exporter.get_unique_id()}'
        python_code = self.recompile()
        import_block = _format_import_block(python_code.globals, exporter.importer)
        module_code = import_block + self.code
        exporter.save_source_string(generated_module_name, module_code)

        if hasattr(self._graph, 'tracer'):
            self.tracer = self._graph.tracer

        dict_without_graph = self.__dict__.copy()
        del dict_without_graph['_graph']
        return (reduce_package_graph_module, (dict_without_graph, generated_module_name))
```
If we *don't* break this reference cycle, we end up pickling the graph along with the tracer, which might be causing this problem.

Reply:
okay that makes sense. I wonder if we should add some basic enforcement to `Graph` so that it just throws if you try to pickle it. I can imagine that custom tracer authors may keep the graph around for whatever reason.